CesiumGeometry/include/CesiumGeometry/IntersectionTests.h

```diff
@@ -99,6 +102,130 @@ class CESIUMGEOMETRY_API IntersectionTests final {
       const glm::dvec3& triangleVertB,
       const glm::dvec3& triangleVertC,
       glm::dvec3& barycentricCoordinates) noexcept;
+
+  /**
+   * @brief Tests if a ray hits a triangle and returns the hit point.
+   *
+   * @param ray The ray.
+   * @param p0 The first vertex of the triangle.
+   * @param p1 The second vertex of the triangle.
+   * @param p2 The third vertex of the triangle.
+   * @param cullBackFaces Ignore triangles that face away from ray. Front faces
+   * use CCW winding order.
+   * @return The point of intersection, or `std::nullopt` if there is no
+   * intersection.
+   */
+  static std::optional<glm::dvec3> rayTriangle(
+      const Ray& ray,
+      const glm::dvec3& p0,
+      const glm::dvec3& p1,
+      const glm::dvec3& p2,
+      bool cullBackFaces = false);
+
+  /**
+   * @brief Tests if an infinite ray hits a triangle and returns the parametric
+   * hit position.
+   *
+   * The parameter t is positive if the intersection point is in front of the
+   * ray origin, negative if it is behind it, or zero if the two points
+   * coincide.
+   *
+   * @param ray The ray.
+   * @param p0 The first vertex of the triangle.
+   * @param p1 The second vertex of the triangle.
+   * @param p2 The third vertex of the triangle.
+   * @param t Returned parametric value, if a hit is detected.
+   * @param cullBackFaces Ignore triangles that face away from ray. Front faces
+   * use CCW winding order.
+   * @return True if a hit occurred, false otherwise.
+   */
+  static std::optional<double> rayTriangleParametric(
+      const Ray& ray,
+      const glm::dvec3& p0,
+      const glm::dvec3& p1,
+      const glm::dvec3& p2,
+      bool cullBackFaces = false);
+
+  /**
+   * @brief Computes the intersection of a ray and an axis aligned bounding box.
+   *
+   * @param ray The ray.
+   * @param aabb The axis aligned bounding box.
+   * @return The point of intersection, or `std::nullopt` if there is no
+   * intersection.
+   */
+  static std::optional<glm::dvec3>
+  rayAABB(const Ray& ray, const AxisAlignedBox& aabb);
+
+  /**
+   * @brief Computes the intersection of an infinite ray and an axis aligned
+   * bounding box and returns the parametric hit position.
+   *
+   * The parameter t is positive if the intersection point is in front of the
+   * ray origin, negative if it is behind it, or zero if the two points
+   * coincide.
+   *
+   * @param ray The ray.
+   * @param aabb The axis aligned bounding box.
+   * @param t Returned parametric value, if a hit is detected.
+   * @return True if a hit occurred, false otherwise.
+   */
+  static bool
+  rayAABBParametric(const Ray& ray, const AxisAlignedBox& aabb, double& t);
```

Comment:
This should also return std::optional, same as `rayTriangleParametric` now does. Same with the other parametric intersection tests below.

Reply:
Fixed up the whole file. Should be t-less now...